packages/serverless-offline-kinesis/src/index.js

```diff
@@ -132,41 +157,59 @@ class ServerlessOfflineKinesis {
     if (typeof streamEvent.arn === 'string') return extractStreamNameFromARN(streamEvent.arn);
     if (typeof streamEvent.streamName === 'string') return streamEvent.streamName;
 
-    if (streamEvent.arn['Fn::GetAtt']) {
-      const [ResourceName] = streamEvent.arn['Fn::GetAtt'];
-
-      if (
-        this.service &&
-        this.service.resources &&
-        this.service.resources.Resources &&
-        this.service.resources.Resources[ResourceName] &&
-        this.service.resources.Resources[ResourceName].Properties &&
-        typeof this.service.resources.Resources[ResourceName].Properties.Name === 'string'
-      )
-        return this.service.resources.Resources[ResourceName].Properties.Name;
+    const getAtt = streamEvent.arn['Fn::GetAtt'];
+    if (getAtt) {
+      const logicalResourceName = extractStreamNameFromGetAtt(getAtt);
+      const physicalResourceName = get(['service', 'resources', 'Resources', logicalResourceName, 'Properties', 'Name'])(this);
+      if (typeof physicalResourceName === 'string')
+        return physicalResourceName;
+    }
+    const join = streamEvent.arn['Fn::Join'];
+    if (join) {
+      const physicalResourceName = extractStreamNameFromJoin(join);
+      if (typeof physicalResourceName === 'string')
+        return physicalResourceName;
     }
 
     throw new Error(
       `StreamName not found. See https://github.com/CoorpAcademy/serverless-plugins/tree/master/packages/serverless-offline-kinesis#functions`
```

Comment:
Would it help to include the unfound stream name in the error?

Reply:
Well the problem is that we couldn't resolve the stream name from the given specification, so we'll have to find a way to print that out.